File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/KeyboardVisibilityHandler.java
package closer.vlllage.com.closer.handler.helpers;

import android.view.View;

import closer.vlllage.com.closer.pool.PoolMember;
import io.reactivex.Observable;
import io.reactivex.subjects.BehaviorSubject;

public class KeyboardVisibilityHandler extends PoolMember {

    private BehaviorSubject<Boolean> isKeyboardVisibleObservable = BehaviorSubject.createDefault(false);
    private int lastKeyboardHeight = 0;

    public void attach(View viewFitsSystemWindows) {
        viewFitsSystemWindows.addOnLayoutChangeListener((v, left, top, right, bottom, oldLeft, oldTop, oldRight, oldBottom) -> {
            if (viewFitsSystemWindows.getPaddingBottom() != 0) {
                lastKeyboardHeight = viewFitsSystemWindows.getPaddingBottom();
            }

            isKeyboardVisibleObservable.onNext(viewFitsSystemWindows.getPaddingBottom() != 0);
        });
    }

    public Observable<Boolean> isKeyboardVisible() {
        return isKeyboardVisibleObservable.distinctUntilChanged();
    }

    public int getLastKeyboardHeight() {
        return lastKeyboardHeight;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/KeyboardVisibilityHandler.kt
package closer.vlllage.com.closer.handler.helpers

import android.view.View

import com.queatz.on.On
import io.reactivex.Observable
import io.reactivex.subjects.BehaviorSubject

class KeyboardVisibilityHandler constructor(private val on: On) {

    private val isKeyboardVisibleObservable = BehaviorSubject.createDefault(false)
    var lastKeyboardHeight = 0
        private set

    val isKeyboardVisible: Observable<Boolean>
        get() = isKeyboardVisibleObservable.distinctUntilChanged()

    fun attach(viewFitsSystemWindows: View) {
        viewFitsSystemWindows.addOnLayoutChangeListener { v, left, top, right, bottom, oldLeft, oldTop, oldRight, oldBottom ->
            if (viewFitsSystemWindows.paddingBottom != 0) {
                lastKeyboardHeight = viewFitsSystemWindows.paddingBottom
            }

            isKeyboardVisibleObservable.onNext(viewFitsSystemWindows.paddingBottom != 0)
        }
    }
}
